assets/js/base/components/cart-checkout/totals/shipping/index.tsx

```diff
@@ -77,6 +81,17 @@ export const TotalsShipping = ( {
 		}
 	);
 
+	const { clearValidationError } = useDispatch( VALIDATION_STORE_KEY );
+	const { removeNotice } = useDispatch( 'core/notices' );
+
+	// Clear shipping rates validation error and notice when local pickup is selected.
+	useEffect( () => {
+		if ( prefersCollection ) {
+			clearValidationError( 'shipping-rates-error' );
+			removeNotice( 'shipping-rates-error', 'wc/checkout' );
+		}
+	}, [ clearValidationError, prefersCollection, removeNotice ] );
+
```

Comment:
This doesn't seem like the right place to trigger this, it should be at the pickup/shipping selector level to ensure it only runs in Checkout and not both Cart and Checkout.